Xamarin.Forms.Core/CarouselView/CarouselView.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Collections;
+
+namespace Xamarin.Forms
+{
+	/// <summary>
+	/// CarouselView Interface
+	/// </summary>
+	public class CarouselView : View //Layout<View>
+	{
+		public static readonly BindableProperty ItemsSourceProperty = BindableProperty.Create("ItemsSource", typeof(IList), typeof(CarouselView), null);
+
+		public IList ItemsSource
+		{
+			get { return (IList)GetValue(ItemsSourceProperty); }
+			set { SetValue(ItemsSourceProperty, value); }
+		}
+
+		public static readonly BindableProperty ItemTemplateProperty = BindableProperty.Create("ItemTemplate", typeof(DataTemplate), typeof(CarouselView), null);
+
+		public DataTemplate ItemTemplate
+		{
+			get { return (DataTemplate)GetValue(ItemTemplateProperty); }
+			set { SetValue(ItemTemplateProperty, value); }
+		}
```

Comment:
Not possible as Xamarin.Forms Views need to be added to native controls in their native representation.